File: algorithms/541. Reverse String II/solution.py
class Solution(object):
    def reverseStr(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: str
        """
        return ''.join(s[i:i+k][::-1] + s[i+k:i+2*k] for i in xrange(0, len(s), 2*k))

File: algorithms/541. Reverse String II/solution.cc
class Solution {
public:
    string reverseStr(string s, int k) {
        for ( int i = 0; i < s.size(); i += 2 * k ) {
            reverse(s.begin() + i, min(s.begin() + i + k, s.end()));
        }
        return s;
    }
};

// 60 / 60 test cases passed.
// Status: Accepted
// Runtime: 9 ms
